core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStoreDir.java

```diff
@@ -100,44 +100,42 @@ public void scanPages(Consumer<PageInfo> pageInfoConsumer) throws IOException {
    * @param path path of a file
    * @return the corresponding page info for the file otherwise null
    */
-  @Nullable
-  private PageInfo getPageInfo(Path path) {
-    PageId pageId = getPageId(path);
-    long pageSize;
-    if (pageId == null) {
-      LOG.error("Unrecognized page file" + path);
-      return null;
-    }
-    try {
-      pageSize = Files.size(path);
-    } catch (IOException e) {
-      LOG.error("Failed to get file size for " + path, e);
-      return null;
-    }
-    return new PageInfo(pageId, pageSize, this);
+  private Optional<PageInfo> getPageInfo(Path path) {
+    return getPageId(path).map(pageId -> {
+      long pageSize;
+      try {
+        pageSize = Files.size(path);
+      } catch (IOException e) {
+        LOG.error("Failed to get file size for " + path, e);
+        return null;
```

Comment:
It's a little weird to still see `return null` here. Maybe the logic is complicated enough it's better not to use `Optional.map`

Reply:
Yeah, I have the same feeling,  I just appended a commit to this PR to removed the usage of return null and Optional.map.  @rongrong  Do you wanna take another look?  Thanks!